metastore/metastore-api/src/main/java/org/apache/drill/metastore/components/tables/BasicTablesTransformer.java

```diff
@@ -99,6 +99,9 @@ public static MetadataHolder all(List<TableMetadataUnit> units) {
         case PARTITION:
           partitions.add(PartitionMetadata.builder().metadataUnit(unit).build());
           break;
+        default:
+          // Throw an error?
```

Comment:
Please either throw an error here or remove this comment. But it is impossible to obtain a non-listed type here.